src/main/java/frc/robot/RobotContainer.java

```diff
@@ -271,19 +309,91 @@ private void configureButtonBindings() {
+
+    /* This is for creating the button mappings for logging what coral have been scored
+     * The Driverstation has a hard limit of 32 buttons so we use 2 different vjoy controllers
+     * to get the effective 64 buttons that we need for logging. this first 16 buttons of every controller are
+     * used for the front and back coral scored poses. */
+    int rows = 3;
+    for (int i = 0; i < Constants.kPoleLetters.length / 2; i++) {
+      for (int j = 0; j < rows; j++) {
+        OI.getButton(OI.StreamDeck.streamDeckButtons[0][i * rows + j])
+            .onTrue(drive.setPoseScored(Constants.kPoleLetters[i], j))
+            .onFalse(drive.setPoseScored(Constants.kPoleLetters[i], j));
+        OI.getButton(OI.StreamDeck.streamDeckButtons[1][i * rows + j])
+            .onTrue(drive.setPoseScored(Constants.kPoleLetters[i + 6], j))
+            .onFalse(drive.setPoseScored(Constants.kPoleLetters[i + 6], j));
+      }
+    }
+
+    // Button to update Setpoints of the elevator based on the Stream Deck nobs
+    OI.getButton(OI.StreamDeck.streamDeckButtons[1][31])
+        .onTrue(
+            elevator
+                .tuneSetpoints(
+                    OI.getAxisSupplier(OI.StreamDeck.Nob1),
+                    () -> DriverStation.getStickAxis(2, 1), // TODO: Figure Out Axis Bug
+                    OI.getAxisSupplier(OI.StreamDeck.Nob3),
+                    OI.getAxisSupplier(OI.StreamDeck.Nob4))
+                .ignoringDisable(true));
+
+    // Temp Keyboard Buttons for sim with no controller
+    if (usingKeyboard) {
+      // Driving
+      drive.setDefaultCommand(
+          DriveCommands.joystickDrive(
+              drive,
+              () -> OI.getAxisSupplier(OI.Keyboard.AD).get(),
+              () -> OI.getAxisSupplier(OI.Keyboard.WS).get(),
+              () -> OI.getAxisSupplier(OI.Keyboard.ArrowLR).get()));
+      OI.getButton(OI.Keyboard.M)
+          .whileTrue(
+              DriveCommands.AlignToReef(
+                  drive,
+                  OI.getAxisSupplier(OI.Keyboard.AD),
+                  OI.getAxisSupplier(OI.Keyboard.WS),
+                  drive.getAlignRotation()));
+
+      // Intake
+      OI.getButton(OI.Keyboard.ForwardSlash).whileTrue(intake.floorIntake());
+
+      // Elevator Buttons
+      OI.getButton(OI.Keyboard.Z).onTrue(elevator.L0());
+      OI.getButton(OI.Keyboard.X).onTrue(elevator.L2());
+      OI.getButton(OI.Keyboard.C).onTrue(elevator.L3());
+      OI.getButton(OI.Keyboard.V).onTrue(elevator.L4());
+
+      // Scorer
+      OI.getButton(OI.Keyboard.Period).whileTrue(coralScorer.scoreCommand());
+    }
+  }
 
-    OI.getPOVButton(OI.Operator.DPAD_LEFT).whileTrue(intake.pivotUpCommand());
-    OI.getPOVButton(OI.Operator.DPAD_RIGHT).whileTrue(intake.pivotDownCommand());
+  public Trigger coralHandoffCompleteTrigger() {
+    return new Trigger(
+        () ->
+            sensors.getSensorState() == CoralEnum.NO_CORAL
+                || coralScorer.hasCoral().getAsBoolean());
```

Comment:
```suggestion
  private Trigger coralHandoffCompleteTrigger  = new Trigger(
        () ->
            sensors.getSensorState() == CoralEnum.NO_CORAL
                || coralScorer.hasCoral().getAsBoolean());
```
Each time this method is called it creates a new Trigger, which isn't necessary. You can just create it once as a private class variable.